File: 1.0/src/cz/mg/vulkan/PFNvkGetDeviceQueue.java
package cz.mg.vulkan;

public class PFNvkGetDeviceQueue extends VkFunctionPointer {
    public PFNvkGetDeviceQueue() {
    }

    protected PFNvkGetDeviceQueue(VkMemory vkmemory) {
        super(vkmemory);
    }

    protected PFNvkGetDeviceQueue(VkMemory vkmemory, long vkaddress) {
        super(vkmemory, vkaddress);
    }



    public PFNvkGetDeviceQueue(long value) {
        setValue(value);
    }

    public PFNvkGetDeviceQueue(VkInstance instance) {
        super(instance, new VkString("vkGetDeviceQueue"));
    }

    public void call(VkDevice device, VkUInt32 queueFamilyIndex, VkUInt32 queueIndex, VkQueue pQueue){
        callNative(getValue(), device != null ? device.getVkAddress() : VkPointer.getNullAddressNative(), queueFamilyIndex != null ? queueFamilyIndex.getVkAddress() : VkPointer.getNullAddressNative(), queueIndex != null ? queueIndex.getVkAddress() : VkPointer.getNullAddressNative(), pQueue != null ? pQueue.getVkAddress() : VkPointer.NULL);
    }

    protected static native void callNative(long vkaddress, long device, long queueFamilyIndex, long queueIndex, long pQueue);


    public void call(VkDevice device, int queueFamilyIndex, int queueIndex, VkQueue pQueue){
        callSimplifiedNative(getValue(), device != null ? device.getVkAddress() : VkPointer.getNullAddressNative(), queueFamilyIndex, queueIndex, pQueue != null ? pQueue.getVkAddress() : VkPointer.NULL);
    }

    protected static native void callSimplifiedNative(long vkaddress, long device, int queueFamilyIndex, int queueIndex, long pQueue);

}

File: 1.0/src/c/cz/mg/vulkan/PFN_vkGetDeviceQueue.c
#include <jni.h>
#include <stdlib.h>
#include <string.h>
#include <vulkan/vulkan.h>

jlong jniPointerToLong(const void* p);
void* jniLongToPointer(jlong l);
jlong jniFunctionPointerToLong(PFN_vkVoidFunction p);
PFN_vkVoidFunction jniLongToFunctionPointer(jlong l);
void jniThrowException(JNIEnv* env, const char* message);

void Java_cz_mg_vulkan_PFNvkGetDeviceQueue_callNative(JNIEnv* env, jclass jc, jlong address, jlong device, jint queueFamilyIndex, jint queueIndex, jlong pQueue){
    (void)env;
    (void)jc;
    PFN_vkGetDeviceQueue f = (PFN_vkGetDeviceQueue)jniLongToFunctionPointer(address);
    f(
        *((VkDevice*)jniLongToPointer(device)),
        (uint32_t)queueFamilyIndex,
        (uint32_t)queueIndex,
        ((VkQueue*)jniLongToPointer(pQueue))
    );
}

